code/modules/mob/living/carbon/human/human_update_icons.dm

```diff
@@ -349,15 +354,30 @@ There are several things that need to be remembered:
 		inv.update_appearance()
 
 	if(belt)
+		//SKYRAT EDIT ADDITION BEGIN - TESHARI CLOTHES
+		var/icon_file = belt.worn_icon
+		var/applied_styles = NONE
+		if(dna.species.id == "teshari")
+			var/list/tesh_icon_states = icon_states('modular_skyrat/master_files/icons/mob/clothing/species/teshari/belt.dmi')
```

Comment:
This one should be static, so it's not generating all the icon states of that file every time it tries to update icons.